include/my_stacktrace.h

```diff
@@ -59,6 +59,24 @@ void my_set_exception_pointers(EXCEPTION_POINTERS *ep);
 void my_write_core(int sig);
 #endif
 
+/* print stack traces for all threads */
+void my_pstack();
+
+/* print stack traces for all threads before assert failures */
+#define assert_with_stack_traces(expr) \
```

Comment:
That's where I was heading, I don't want to mix some large, server-wide change with something that is a necessity to get MyRocks to compile if I can help it. I don't think changing the default global behavior is quite as easy as it seems as every system/engine has its own wrapper for dealing with debug vs release/ship assertions. Then we have the submodule kv libraries that are still behaving independently (PerconaFT, RocksDB) and don't use anything from the upper layers of the server.